src/Farmer/Arm/SignalR.fs

```diff
@@ -4,7 +4,19 @@ module Farmer.Arm.SignalRService
 open Farmer
 open Farmer.SignalR
 
-let signalR = ResourceType("Microsoft.SignalRService/signalR", "2018-10-01")
+let signalR = ResourceType("Microsoft.SignalRService/signalR", "2022-02-01")
+
+type SignalRFilterPattern =
+    | Any
+    | List of string list
+
+type UpstreamTemplate =
+    {
+        UrlTemplate: string
+        HubPattern: SignalRFilterPattern
+        CategoryPattern: SignalRFilterPattern
+        EventPattern: SignalRFilterPattern
+    }
```

Comment:
You could add the Arm encoding here to prevent repeating it
```
    member this.ArmValue = 
        match this with
        | Any -> "*"
        | List values -> String.concat "," values
```